pkg/server/endpoints/node/handler.go

```diff
@@ -219,7 +218,66 @@ func (h *Handler) FetchX509SVID(server node.Node_FetchX509SVIDServer) (err error
 }
 
 func (h *Handler) FetchJWTSVID(ctx context.Context, req *node.FetchJWTSVIDRequest) (*node.FetchJWTSVIDResponse, error) {
-	return nil, errors.New("not implemented")
+	peerCert, err := h.getCertFromCtx(ctx)
+	if err != nil {
+		h.c.Log.Error(err)
+		return nil, errors.New("node SVID is required for this request")
+	}
+
+	// validate request parameters
+	switch {
+	case req.Jsr == nil:
+		return nil, errors.New("request missing JSR")
+	case req.Jsr.SpiffeId == "":
+		return nil, errors.New("request missing SPIFFE ID")
+	case len(req.Jsr.Audience) == 0:
+		return nil, errors.New("request missing audience")
+	}
+
+	agentID, err := getSpiffeIDFromCert(peerCert)
+	if err != nil {
+		h.c.Log.Error(err)
+		return nil, err
+	}
+
+	dataStore := h.c.Catalog.DataStores()[0]
+	regEntries, err := regentryutil.FetchRegistrationEntries(ctx, dataStore, agentID)
+	if err != nil {
+		return nil, err
+	}
+
+	var regEntry *common.RegistrationEntry
+	for _, candidateEntry := range regEntries {
+		if candidateEntry.SpiffeId == req.Jsr.SpiffeId {
```

Comment:
since we don't really use regEntry, the `found` pattern makes a little more sense. i think we also need to allow the case where agentID == req.Jsr.SpiffeId

EDIT: see below comment for (non)applicability

Reply:
👍